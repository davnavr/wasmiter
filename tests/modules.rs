#[test]
fn basic_module() {
    let wat = r#"(module
    (func (export "add_five") (param i32) (result i32)
        local.get 0
        i32.const 5
        i32.add))
"#;
    let wasm = wat::parse_str(wat).unwrap();
    insta::assert_display_snapshot!(&wasmiter::parse_module_sections(wasm.as_slice()).unwrap());
}

#[test]
fn exception_handling() {
    let wasm = wat::parse_str(include_str!("modules/exception_handling.wat")).unwrap();
    insta::assert_display_snapshot!(&wasmiter::parse_module_sections(wasm.as_slice()).unwrap());
}

#[test]
fn all_the_things() {
    let wasm = wat::parse_str(include_str!("modules/all_the_things.wat")).unwrap();
    insta::assert_display_snapshot!(&wasmiter::parse_module_sections(wasm.as_slice()).unwrap());
}
