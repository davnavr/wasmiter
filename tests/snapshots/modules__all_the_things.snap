---
source: tests/modules.rs
expression: module.display_module()
---
(module
(type (; 0 ;) (func (param i32) (result)))
(type (; 1 ;) (func (param) (result)))
(type (; 2 ;) (func (param) (result i32)))
(type (; 3 ;) (func (param i32 i32) (result i32)))

(import "sample_api" "_handles" (table (; 0 ;) 0 externref))
(import "sample_api" "_handles_start" (global (; 0 ;) (mut i32)))
(import "sample_api" "panic" (tag (; 0 ;) (type 0)))
(import "extern" "memory" (memory (; 0 ;) 16))
(import "extern" "fun_startup" (func (; 0 ;) (type 1)))
(import "extern" "fun_link_startup" (func (; 1 ;) (type 2)))

;; function section count = 3
(table (; 0 ;) 0 funcref)

(memory 0 16)

(global (mut i32) i32.const 0x00000000 (; 0 signed, 0 unsigned ;) )
(global (mut i32) i32.const 0x00000000 (; 0 signed, 0 unsigned ;) )

(export "_start" (func 3))

(start 4)
(func (type 3) ;; code size = 4
  i32.const 0x00000000 (; 0 signed, 0 unsigned ;)
)
(func (type 1) ;; code size = 10
  i32.const 0xD81C1C1C (; -669246436 signed, 3625720860 unsigned ;)
  drop
  nop
)
(func (type 1) ;; code size = 5
  call 1
  drop
)

(data (; 0 ;) "el amor")
(data (; 1 ;)
  "\FF\FF\00QQQQQQQQQQQQQ"
  "Q\07")

(; UNRECOGNIZED (0) @ 0x112 to 0x176
[
    046E616D_65012304__010B6675_6E5F7374,
    61727475_7002046D__61696E03_065F7374,
    61727404_055F696E__69740512_0200085F,
    48414E44_4C455301__0546554E_4353060A,
    01010753_43524154__43480719_02010946,
    554E4353_5F505452__020B5343_52415443,
    485F5054_52,
]
;)
)
